OpenRA.Mods.Common/Traits/TooltipDescription.cs

```diff
@@ -20,15 +20,33 @@ public class TooltipDescriptionInfo : ConditionalTraitInfo
 		[Translate]
 		public readonly string Description = "";
 
-		public override object Create(ActorInitializer init) { return new TooltipDescription(this); }
+		[Desc("Player stances who can view the description.")]
+		public readonly Stance ValidStances = Stance.Ally | Stance.Neutral | Stance.Enemy;
+
+		public override object Create(ActorInitializer init) { return new TooltipDescription(init.Self, this); }
 	}
 
 	public class TooltipDescription : ConditionalTrait<TooltipDescriptionInfo>, IProvideTooltipInfo
 	{
-		public TooltipDescription(TooltipDescriptionInfo info)
-			: base(info) { }
+		readonly Actor self;
+
+		public TooltipDescription(Actor self, TooltipDescriptionInfo info)
+			: base(info)
+		{
+			this.self = self;
+		}
 
-		public bool IsTooltipVisible(Player forPlayer) { return !IsTraitDisabled; }
+		public bool IsTooltipVisible(Player forplayer)
+		{
+			var stance = forplayer.Stances[self.Owner];
+			if (!Info.ValidStances.HasStance(stance))
+				return false;
+
+			if (IsTraitDisabled)
```

Comment:
I think it'd be better to move this check to the first line in the function, so we save the stance lookup if the trait is disabled.

Reply:
Will correct when I'm home.

Reply:
Should `var` also be moved?